tests/queries/0_stateless/02677_analyzer_compound_expressions.sql

```diff
@@ -0,0 +1,44 @@
+SET allow_experimental_analyzer = 1;
+
+WITH ('a', 'b')::Tuple(c1 String, c2 String) AS t
+SELECT t.c1, t.c2;
+
+WITH materialize(('a', 'b')::Tuple(c1 String, c2 String)) AS t
+SELECT t.c1, t.c2;
+
+WITH (1, ('a', 'b'))::Tuple(c1 UInt64, t1 Tuple(c1 String, c2 String)) AS t
+SELECT t.c1, t.t1.c1, t.t1.c2;
+
+WITH materialize((1, ('a', 'b'))::Tuple(c1 UInt64, t1 Tuple(c1 String, c2 String))) AS t
+SELECT t.c1, t.t1.c1, t.t1.c2;
+
+WITH [1, 2, 3] AS arr SELECT arr.size0;
+WITH materialize([1, 2, 3]) AS arr SELECT arr.size0;
+
+WITH [1, 2, NULL] AS arr SELECT arr.null;
+WITH materialize([1, 2, NULL]) AS arr SELECT arr.null;
+
+WITH [[1, 2], [], [3]] AS arr SELECT arr.size0, arr.size1;
+WITH materialize([[1, 2], [], [3]]) AS arr SELECT arr.size0, arr.size1;
+
+WITH map('foo', 1, 'bar', 2) AS m SELECT m.keys, m.values;
+WITH materialize(map('foo', 1, 'bar', 2)) AS m SELECT m.keys, m.values;
+
+WITH map('foo', (1, 2), 'bar', (3, 4))::Map(String, Tuple(a UInt64, b UInt64)) AS m
+SELECT m.keys, m.values, m.values.a, m.values.b;
+
+WITH materialize(map('foo', (1, 2), 'bar', (3, 4))::Map(String, Tuple(a UInt64, b UInt64))) AS m
+SELECT m.keys, m.values, m.values.a, m.values.b;
+
+WITH map('foo', (1, 2), 'bar', (3, 4))::Map(String, Tuple(a UInt64, b UInt64)) AS m
+SELECT m.keys, m.values, m.values.*;
+
+WITH materialize(map('foo', (1, 2), 'bar', (3, 4))::Map(String, Tuple(a UInt64, b UInt64))) AS m
+SELECT m.keys, m.values, m.values.*;
+
+WITH [1, 2, 3] AS arr SELECT arr.*; -- { serverError UNSUPPORTED_METHOD }
+WITH map('foo', 1, 'bar', 2) AS m SELECT m.*; -- { serverError UNSUPPORTED_METHOD }
```

Comment:
Not sure if this intended ? If there are subcolumns we should be able to match them using matchers, then it will be natural.

Reply:
I think it makes sense only for `Map` type. For `Array` and `Nullable` subcolumns are kind of aliases and aliases are usually not included to asterisk matcher.

Reply:
Make sense.